engine/src/main/java/org/terasology/entitySystem/event/internal/EventSystemImpl.java

```diff
@@ -241,6 +241,16 @@ private void addToComponentSpecificHandlers(Class<? extends Event> type, EventHa
                 }
             }
         }
+
+        if (0 == componentTypes.length) {
+            Iterator<EventHandlerInfo> eventHandlerIterator = generalHandlers.values().iterator();
+            while (eventHandlerIterator.hasNext()) {
```

Comment:
A little bird named IntelliJ whispered in my ear that this could be replaced with a magic incantation going `Collection.removeIf ...` - neat!

That doesn't get in the way of merging this though, but I thought it was a cool suggestion. Feel free to look for things like that if you're in a class anyway. Same goes for updating those annoying empty javadoc blocks, especially since you doc so well! :D